lib/group_site_manager.py

```diff
@@ -154,6 +156,44 @@ def _load_page(self, driver, url) -> None:
     driver.get(url)
     time.sleep(3)
 
+  def _upload_bfmr(self, numbers) -> None:
+    group_config = self.config['groups']['bfmr']
+    driver = self.driver_creator.new()
+    try:
+      # load the login page first
+      self._load_page(driver, "https://buyformeretail.com/login")
+      driver.find_element_by_id("loginEmail").send_keys(group_config['username'])
+      driver.find_element_by_id("loginPassword").send_keys(
+          group_config['password'])
+      driver.find_element_by_xpath("//button[@type='submit']").click()
+
+      time.sleep(2)
+
+      # hope there's a button to submit tracking numbers -- it doesn't matter which one
+      submit_button = driver.find_element_by_xpath(
```

Comment:
Have error handling in case one isn't found?  It's fine to abort at that point, just have a coherent error message.

Also, this could fail either because login fails or because there isn't an available textbox.  Hopefully there's some way for those to be two separate checks.

Reply:
yeah i can just catch/raise here

I think there's an issue for better login failure messages across the board